app.py

```
@@ -0,0 +1,188 @@
+import json
+import os
+import random
+from typing import Dict, List, Tuple
+
+import requests
+from flask import Flask, request
+from newsdataapi import NewsDataApiClient
+
+
+def send_newsapi_request(keyword: str) -> Tuple[Dict, int]:
+    api_key = os.getenv("NEWSAPI_KEY")
+    params = {"q": keyword, "sortBy": "publishedAt", "apiKey": api_key}
+    response = requests.get(
+        url="https://newsapi.org/v2/everything", params=params, timeout=5
+    )
+    if response.ok:
+        return json.loads(response.content.decode("utf-8")), response.status_code
+    return {}, response.status_code
+
+
+def parse_newsapi_response(newsapi_response: Dict) -> Dict:
+    response = {"totalResults": newsapi_response["totalResults"], "results": []}
+    for news in newsapi_response["articles"]:
+        response["results"].append(
+            {
+                "semanticInfo": {},
+                "source": news["source"],
+                "author": news["author"],
+                "title": news["title"],
+                "description": news["description"],
+                "url": news["url"],
+                "urlToImage": news["urlToImage"],
+                "publishedAt": news["publishedAt"],
+                "content": news["content"],
+            }
+        )
+    return response
+
+
+def send_newsdataapi_request(keyword: str) -> Tuple[Dict, int]:
+    api_key = os.getenv("NEWSDATAAPI_KEY")
+    api = NewsDataApiClient(apikey=api_key)
+    response = api.news_api(q=keyword, country="us")
+    if response["status"] != "success":
+        return response, 400

```

You should propagate the error messages to the client if you do not deal with it. One possible way to do this is forward the serialized response from `NewsDataApiClient` to front-end website, so the debugging/error message is not lost.

---

Will be applied in refactor branch